ft_scalpcurrentdensity.m

```diff
@@ -185,24 +185,32 @@
 
 allchanpos = elec.chanpos(elecindx,:);    % the position of all channels, ordered according to the data
 goodchanpos = allchanpos(goodindx,:);     % the position of good channels
-anychanpos  = [0 0 1];                    % random channel, will be used in case there are no bad channels
 
 % compute SCD for each trial
 if strcmp(cfg.method, 'spline')
+  
+  fprintf('Checking spherical fit... ');
+  [c, r] = fitsphere(chanpos);
+  d = chanpos - repmat(c, numel(find(sensidx)), 1);
+  d = sqrt(sum(d.^2, 2));
+  d = mean(abs(d) / r);
+  if abs(d-1) > 0.1
+    ft_warning('bad spherical fit (residual: %.2f%%). The interpolation will be inaccurate.', 100*(d-1));
+  elseif abs(d-1) < 0.01
+    fprintf('perfect spherical fit (residual: %.1f%%)\n', 100*(d-1));
+  else
+    fprintf('good spherical fit (residual: %.1f%%)\n', 100*(d-1));
+  end
+  
+  % Builds the spatial filter only once.
+  fprintf('Calculating the filter to build the SCD.\n');
+  [WVo, WLo] = sphsplint(goodchanpos, allchanpos, cfg.order, cfg.degree, cfg.lambda);
+  
   ft_progress('init', cfg.feedback, 'computing SCD for trial...')
   for trlop=1:Ntrials
     % do compute interpolation
     ft_progress(trlop/Ntrials, 'computing SCD for trial %d of %d', trlop, Ntrials);
-    if ~isempty(cfg.badchannel)
-      % compute scd for all channels, also for the bad ones
-      fprintf('computing scd also at locations of bad channels');
-      [V2, L2, L1] = splint(goodchanpos, data.trial{trlop}(goodindx,:), allchanpos, cfg.order, cfg.degree, cfg.lambda);
-      scd.trial{trlop} = L2;
-    else
-      % just compute scd for input channels, specify arbitrary channel to-be-discarded for interpolation to save computation time
-      [V2, L2, L1] = splint(goodchanpos, data.trial{trlop}(goodindx,:), anychanpos, cfg.order, cfg.degree, cfg.lambda);
-      scd.trial{trlop} = L1;
-    end
+    scd.trial{trlop} = WLo * data.trial{trlop}(goodindx,:);
```

Comment:
I think that at this point we could consider using ft_apply_montage, both to the data object, and to the electrode description. I know that it is not part of this PR per se, but for code consistency w.r.t. the other methods it would be nice to have it. I can also imagine that we keep this for later

Reply:
I think it is a good idea, as it would also solve part of the issues I found with the Hjorth method (see https://github.com/fieldtrip/fieldtrip/issues/1841#issuecomment-893580074).

I will change it.

Reply:
Done.

Should we use ft_apply_montage too in ft_channelrepair? All methods but 'nan' use a spatial filter to build the interpolated data from the original.